net/respondd/src/respondd.c

```diff
@@ -183,11 +183,20 @@ static const struct respondd_provider_info * get_providers(const char *filename)
 	snprintf(path, sizeof(path), "./%s", filename);
 
 	void *handle = dlopen(path, RTLD_NOW|RTLD_LOCAL);
-	if (!handle)
+	if (!handle) {
+		syslog(LOG_WARNING, "unable to open provider module '%s', ignoring: %s", path, dlerror());
 		return NULL;
+	}
+
+	// clean a potential previous error
+	dlerror();
 
 	const struct respondd_provider_info *ret = dlsym(handle, "respondd_providers");
 	if (!ret) {
+		char *error = dlerror();
+		if (error) {
+			syslog(LOG_WARNING, "unable to load providers from '%s', ignoring: %s", path, error);
+		}
```

Comment:
I think the message should be logged whenever dlsym() returns NULL, not only when dlerror() returns something. So either we could just pass `dlerror() ?: "whatever error message we should use here..."`, or, even simpler, don't use dlerror() at all for the second message, as it wouldn't give a much more specific error message than "unable to load providers from '%s'" already does.

Also, use *filename* instead of *path* for the error messages.